twindb_infrastructure/providers/aws.py

```diff
@@ -207,71 +206,81 @@ def launch_ec2_instance(instance_profile, region=AWS_REGIONS[0],
             mount_volumes(ip, private_key_file, username,
                           volumes=instance_profile['BlockDeviceMappings'])
         return instance_id
-    except OSError as err:
+    except AwsError as err:
         log.error(err)
         return None
 
 
 def add_name_tag(instance_id, name):
-    cmd = ["aws", "ec2", "create-tags",
-           "--output", "json",
-           "--resources", instance_id,
-           "--tags", "Key=Name,Value=%s" % name
-           ]
+    """
+    Added tag to instance
+    :param instance_id: Instance id
+    :param name: Tag for instance
+    :type instance_id: str
+    :type name: str
+    :raise: AwsError
+    :return: Result of setting tag
+    :rtype: bool
+    """
     try:
-        aws_process = Popen(cmd, stdout=PIPE, stderr=PIPE)
-        cout, cerr = aws_process.communicate()
+        client = boto3.client('ec2')
+        response = client.create_tags(
+            Resources=[instance_id],
+            Tags=[
+                {
+                    'Key': 'Name',
+                    'Value': name
+                }
+            ]
+        )
+        return response[0]['ResponseMetadata']['HTTPStatusCode'] == 200
+    except ClientError as err:
+        raise AwsError(err)
+    except ValueError as err:
+        raise AwsError('Internal bug in boto3: %s', err)
 
-        if aws_process.returncode != 0:
-            log.error(cerr)
-            return False
 
-        return True
-    except OSError as err:
-        log.error(err)
-        return False
+def associate_address(instance_id, public_ip=None, private_ip=None):
+    """
+    Associates an Elastic IP address with an instance
 
+    :param instance_id: Instance id
+    :param public_ip: Public ip
+    :param private_ip: Private ip
+    :type instance_id: str
+    :type public_ip: str
+    :type private_ip: str
+    :raise: AwsError
+    :return: Result of associates
+    :rtype: bool
```

Comment:
We won't analyze the return code, so let's return nothing and raise `AwsError` if any errors.